### PR TITLE
augeas: add recursive search & non-standard config locations

### PR DESCRIPTION
**Pull request by examon**
Work in progress on improving our augeas actor.

Run augeas actor as:

`snactor_runner --actors_dir src/actors/ --schemas_dir src/schemas/ augeas <<< '{"aug_input": [{"lens": "httpd", "directives": ["Include", "IncludeOptional"], "load_files": ["/opt/some_nonstandard_config.conf"], "prefix_for_relative": "/etc/httpd"}]}'`

Which now tells augeas actor that is should:
- take a closer look at httpd lens
- take all `Include` and `IncludeOptional` directives from configs and recursively process their arguments
- load into httpd lens `/opt/some_nonstandard_config.conf` file
- use prefix `/etc/httpd` for all file paths that are not absolute

If you want to only tell augeas to load some non-standard files and don't care about processing directives and stuff, use:

`snactor_runner --actors_dir src/actors/ --schemas_dir src/schemas/ augeas <<< '{"aug_input": [{"lens": "httpd", "load_files": ["/opt/test_httpd.conf"]}]}'`

Requirements:
- `augeas`
- `augeas-libs`
- `python-augeas`

### CODE REVIEW

File: src/actors/common/augeas/actor.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 import fnmatch
 from collections import namedtuple, OrderedDict
 
+import magic
```
Comment (shaded-enmity) [inline src/actors/common/augeas/actor.py]: does this work when running the actor remotely?

File: src/actors/common/augeas/actor.py [hunk 2]
```diff
@@ -27,7 +29,7 @@ def get_lenses():
         3) Excluded paths
            Currently unused
     '''
-    augeas = subprocess.check_output(["augtool", "--include=lenses", "dump-xml", "/augeas"])
+    augeas = subprocess.check_output(["augtool"] + magic.do_magic() + ["--include=lenses", "dump-xml", "/augeas"])
```
Comment (shaded-enmity) [inline src/actors/common/augeas/actor.py]: Can we go for something like `AugeasHelper.build()`? At first I thought this has something to do with `libmagic`.

File: src/actors/common/augeas/magic.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import shlex
+import subprocess
+import uuid
+
+from pprint import pprint as pp
+
+import augeas
+
+## FIXME: this is tmp, make this as optional actor input
+INPUT = {"load": ["/opt/test_httpd.conf"],
```
Comment (shaded-enmity) [inline src/actors/common/augeas/magic.py]: what is the purpose of this `INPUT` structure? IIUIC it specifies what additional files to `load`, which `directives` should be used for recursive scanning, and for which `lens` this recursive scanning should take place? How would I use this to add support for, e.g. postgresql `pg_hba.conf`?

File: src/actors/common/augeas/magic.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            ## Locate all DIRECTIVE keywords in config and get its location in augeas tree.
+            directive_arg_paths = augeas_get_directive_argument(augeas_file_path, directive)
+
+            ## Get values for all located directives/
+            for path in directive_arg_paths:
+                directive_arg = augeas_get_path_value(path)
+                #print("FOUND!:", augeas_file_path, directive, directive_arg)
+                if not directive_arg.startswith('/'):
+                    ## FIXME: get root from non-hardcoded stuff
+                    directive_arg = "/etc/httpd/" + directive_arg
```
Comment (shaded-enmity) [inline src/actors/common/augeas/magic.py]: Why do we need to be `httpd` specific here?

**Comment by shaded-enmity** [comment]
The changes look solid, however merging this would make all CI jobs fail, so I guess that we need to update `.travis.yml` first, so that we use fedora/centos env OR install https://launchpad.net/ubuntu/+source/augeas - of these two, I think that using fedora/centos is the better way to go.

**Comment by shaded-enmity** [comment]
@examon ha, judging by the build log we also need to install `gcc` and supposedly `redhat-rpm-config` as well.

**Comment by examon** [comment]
I had to change centos6 to centos7. 
When using centos6, I get 
`setuptools must be installed to install from a source distribution`
when installing pip packages. (setuptools pkg is installed btw)
When upgeaded pip to newer version, I get this:
`setuptools requires Python '>=2.7,!=3.0.*,!=3.1.*,!=3.2.*' but the running Python is 2.6.6`

Centos7 has python 2.7 and it works.

**Pull request merged**